### PR TITLE
i1546 - Show list of report readers in sidebar

### PR DESCRIPTION
**Pull request by littledice**
For users with the `*/roles.read` permission, show a list of users who are able to read the report.

But first review/merge 
https://github.com/vimc/montagu-db/pull/64
and
https://github.com/vimc/montagu-api/pull/249

and point db/api versions back to master

### CODE REVIEW

File: app/src/main/report/components/Reports/Sidebar.tsx [hunk 1]
```diff
@@ -1,97 +0,0 @@
-import * as React from "react";
```
Comment (littledice) [inline app/src/main/report/components/Reports/Sidebar.tsx]: Moved to `/Sidebar` subdirectory and split into 2 components

File: app/src/main/report/components/Sidebar/UserList.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as React from "react"
```
Comment (littledice) [inline app/src/main/report/components/Sidebar/UserList.tsx]: Right now this component is very simple, but in a future task we will add functionality to add/remove users from the list

File: app/src/test/report/reducers/ReportsReducersTests.ts [hunk 3]
```diff
@@ -1,31 +0,0 @@
-import { expect } from "chai";
```
Comment (littledice) [inline app/src/test/report/reducers/ReportsReducersTests.ts]: Noticed that we had 2 files of report reducer tests. Have moved these 4 tests into the other file.

File: app/src/main/report/components/Sidebar/SidebarAdmin.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    return <div>
+        <ReportVersionSwitcher
+            currentVersion={props.version}
+            versions={props.allVersions}
+            onChangeVersion={props.onChangeVersion}
+        /> {props.isReviewer && <PublishSwitch name={props.report}
+                                               version={props.version}
+                                               published={props.published}/>}
+        {props.isAdmin &&
```
Comment (littledice) [inline app/src/main/report/components/Sidebar/SidebarAdmin.tsx]: This is the new bit

File: app/src/main/report/services/ReportsService.ts [hunk 5]
```diff
@@ -3,17 +3,17 @@ import {AbstractReportLocalService} from "./AbstractReportLocalService";
...
 
     getReportVersions(reportId: string) {
-        return this.setOptions({cache: ReportsCacheKeysEnum.versions})
+        return this.setOptions({cacheKey: ReportsCacheKeysEnum.versions})
             .get(`/reports/${reportId}/`);
     }
 
     getVersionDetails(reportId: string, versionId: string) {
-        return this.setOptions({cache: ReportsCacheKeysEnum.versionDetails})
+        return this.setOptions({cacheKey: ReportsCacheKeysEnum.versionDetails})
```
Comment (littledice) [inline app/src/main/report/services/ReportsService.ts]: Just a rename for clarity

File: app/src/test/report/actions/UserActionTests.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        sandbox.restore();
+    });
+
+    it("dispatches report readers fetched action if get report readers action is dispatched", (done) => {
+        const fakeUser = mockUser();
+        sandbox.setStubFunc(UserService.prototype, "getReportReaders", () => {
+            return Promise.resolve([fakeUser]);
+        });
+        store.dispatch(userActions.getReportReaders("test"));
+        setTimeout(() => {
```
Comment (MartinEden) [inline app/src/test/report/actions/UserActionTests.ts]: You should use `checkAsync` here - it handles calling `done` for you, even in the case where an exception is thrown, where this code will just hang.

**Comment by littledice** [comment]
I know, sorry for length :( quite a few incidental changes crept in

**Pull request merged**